### PR TITLE
Questionnaire Refactoring

### PR DESCRIPTION
**Pull request by pallab-gain**
More details
- https://github.com/somia/mobile/milestone/2

### CODE REVIEW

File: ninchatsdk/src/main/java/com/ninchat/sdk/ninchatquestionnaire/ninchatquestionnairelist/presenter/NinchatQuestionnaireListPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    override fun onCompleteLoading(target: String?, thankYouText: String?, loaded: Boolean, position: Int) {
+        model.answerList.getOrNull(position)?.apply {
+            putOpt("loaded", true)
+        }
+    }
+
+    open fun init() {
+        TODO("implement me")
```
Comment (plokk) [inline ninchatsdk/src/main/java/com/ninchat/sdk/ninchatquestionnaire/ninchatquestionnairelist/presenter/NinchatQuestionnaireListPresenter.kt:113]: What about these?

File: ninchatsdk/src/main/res/layout/checkbox_simple.xml [hunk 2]
```diff
@@ -3,16 +3,24 @@
...
+
+    <TextView
+        android:id="@+id/ninchat_checkbox_label"
+        android:layout_marginStart="10dp"
+        android:layout_gravity="center"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_weight="0.9"
+        android:textColor="@color/ninchat_color_text_normal"
+        android:text="A compound text box label "/>
```
Comment (plokk) [inline ninchatsdk/src/main/res/layout/checkbox_simple.xml:24]: Should these be translated?

File: ninchatsdk/src/main/java/com/ninchat/sdk/ninchatquestionnaire/ninchatquestionnairelist/presenter/NinchatQuestionnaireListPresenter.kt (hunk 1 above)
Comment (pallab-gain) [reply ninchatsdk/src/main/java/com/ninchat/sdk/ninchatquestionnaire/ninchatquestionnairelist/presenter/NinchatQuestionnaireListPresenter.kt:113]: They are intentional. :D 
They are methods of an abstract class. In kotlin TODO is `NotImplementedError`.https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-t-o-d-o.html

It is excepted that the class that implementing this abstract class, must implement those methods. There is no particular reason why I follow this pattern ( just a better note that these methods must be implemented by child class ). It seems I inherited this practice from some other projects.

File: ninchatsdk/src/main/res/layout/checkbox_simple.xml (hunk 2 above)
Comment (pallab-gain) [reply ninchatsdk/src/main/res/layout/checkbox_simple.xml:24]: Yes, the translation part is done through checkbox data model class [model class ](https://github.com/somia/ninchat-sdk-android/blob/feature/questionnaire-refactoring/ninchatsdk/src/main/java/com/ninchat/sdk/ninchatquestionnaire/ninchatcheckbox/model/NinchatCheckboxViewModel.kt#L22)

It seems the comment in line 26 is wrong ( copied from button data model class ). I will update the comment

**Pull request merged**